Providers/RustSyncManager.swift

```diff
@@ -0,0 +1,628 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Account
+import Shared
+import Storage
+import Sync
+import SyncTelemetry
+import AuthenticationServices
+import Common
+
+// Extends NSObject so we can use timers.
+public class RustSyncManager: NSObject, SyncManager {
+    // We shouldn't live beyond our containing BrowserProfile, either in the main app
+    // or in an extension.
+    // But it's possible that we'll finish a side-effect sync after we've ditched the
+    // profile as a whole, so we hold on to our Prefs, potentially for a little while
+    // longer. This is safe as a strong reference, because there's no cycle.
```

Comment:
So this was defintely copied as is so I'd defer to @nbhasin2 on this one, but now that we've removed `unowned` is anything in the comment accurate? Because now we're holding on to prefs and profile which seems to negate the premise of the comment.

And to amplify Tarik's point about the implications, are we comfortable with the effects of removing `unowned` here?

Reply:
Is all you are trying to do is access `Profile`?

Reply:
Yeah I want some of the member functions and properties of the given profile.

Reply:
When once you have that initialization ready in another PR we can double-check to make sure if this will have a strong reference cycle or not. A class alone won't have any issues.